numpy/_core/numeric.py

```diff
@@ -2446,14 +2456,24 @@ def array_equal(a1, a2, equal_nan=False):
     if a1.shape != a2.shape:
         return False
     if not equal_nan:
-        return bool(asarray(a1 == a2).all())
+        return bool((a1 == a2).all())
+    cannot_have_nan = _dtype_cannot_hold_nan(a1.dtype)\
+            and _dtype_cannot_hold_nan(a2.dtype)
```

Comment:
```suggestion
    cannot_have_nan = (_dtype_cannot_hold_nan(a1.dtype)
                       and _dtype_cannot_hold_nan(a2.dtype)
```
Just avoid the `\` because I hate it :).

Reply:
👍